aips/aip-124.md

```diff
@@ -0,0 +1,199 @@
+This is done using [derivable account abstraction (AIP-113)](https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-113.md).
+
+## High-level Overview
+
+Allow Sui accounts (Sui Ed25519 key pairs) to create Aptos transactions,
+and be able to authenticate and execute them directly on the Aptos blockchain, without interaction with Sui Blockchain.
+We do so by utilizing ["Sign Personal Message"](https://docs.sui.io/standards/wallet-standard#implementing-features) specification, and defining
+a message format that gives enough details to users so they understand what they are signing reasonably well.
+
+## Impact
+
+This will allow users with Sui Wallets to be able to interact with Aptos blockchain
+
+## Alternative Solutions
+
+Alternatively, wallets from other chains can implement integration with Aptos natively, providing more seamless experience to the users.
+
+## Specification and Implementation Details
+
+dApp is required to issue a [signPersonalMessage](https://docs.sui.io/standards/wallet-standard#implementing-features) request to the wallet with:
+
+```
+<domain> wants you to sign in with your Sui account:
+<sui_address>
+
+Please confirm you explicitly initiated this request from <domain>. You are approving to execute transaction <entry_function> on Aptos blockchain (<network_name>).
+
+Nonce: <aptos_txn_digest>
+```
+
+And submit transaction to Aptos with Authenticator:
+
+```
+AccountAuthenticator::Abstraction {
+    function_info: FunctionInfo::from("0x1::sui_derivable_account::authenticate"),
+    auth_data: AbstractionAuthData::DerivableV1 {
+        signing_message_digest: aptos_txn_digest,
+        abstract_signature: bcs::to_bytes(SuiAbstractSignature::SuiDerivedSignature {
+            signature: sign_personal_message_signature,
+        }),
+        abstract_public_key: bcs::to_bytes(SuiAbstractPublicKey {
+            sui_account_address,
+            domain,
+        }),
+    },
+}
+```
+
+with types:
+
+```
+enum SuiAbstractSignature has drop {
+  SuiDerivedSignature {
+      signature: vector<u8>,
+  },
+}
+
+struct SuiAbstractPublicKey has drop {
+  sui_account_address: vector<u8>,
+  domain: vector<u8>,
+}
+```
+
+And above message will be authorized with `0x1::sui_derivable_account::authenticate`,
+for Aptos account with address derived from `function_info` and `abstract_public_key` with `account_abstraction::derive_account_address`.
+
+`0x1::sui_derivable_account::authenticate` will generate the required text message format,
+extract the `public_key` which is part of the `sign_personal_message_signature`,
+derive the `account_address` from the `public_key`, and verify that the `account_address` matches with the one from the `abstract_public_key`.
+In addition, it will verify that `sign_personal_message_signature` from `abstract_signature` is
+a correct signature of that message.
+
+Looking at what each field in the message is and what is it used for:
```

Comment:
I think you are overloading *message* here.

What is it referring to exactly? 

You should always aim to call things by one (unambiguous) name and one name only. And you should never use the same term to refer to different things: it will cause confusion.